src/core/matchers/toEqual.js

```diff
@@ -11,9 +12,126 @@ getJasmineRequireObj().toEqual = function() {
 
         result.pass = util.equals(actual, expected, customEqualityTesters);
 
+        if (!result.pass) {
+          result.message = buildMessage(actual, expected);
+        }
+
         return result;
       }
     };
+
```

Comment:
One concern that I have is that maybe `buildMessage` and its helper functions should be moved into a separate file. I didn't extract functions as aggressively as I otherwise would have because I wasn't comfortable with the amount of nesting that it produced when everything was in this file. However, I wasn't sure where else to put this functionality — maybe just in another file in `matchers`? I'd like to get the project maintainers' opinion on this.